addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5187,3 +5187,48 @@ def test_cash_basis_with_analytic_distribution_analytic_tax(self):
             },
         ]
         self.assertRecordValues(caba_move.line_ids.sorted('id').sorted('sequence'), expected_caba_move_line_values)
+
+    def test_partial_payments_auto_validation(self):
+        def create_move_payment(move, payment_amount):
+            payment = self.init_payment(amount=payment_amount, post=True, currency=move.currency_id)
+            payment.invoice_ids |= move
+            self.assertEqual(payment.state, 'in_process')
+            return payment
+
+        def reconcile_move(move, transaction_amount, balance=None, date='2023-09-30'):
+            move_line = move.line_ids.filtered(lambda l: l.account_id.account_type in ('asset_receivable', 'liability_payable'))[0]
+            rec_account = move_line.account_id
+            rec_line = self.create_line_for_reconciliation(-(balance or transaction_amount), -transaction_amount, move.currency_id, date, account_1=rec_account)
+            amls = rec_line + move_line
+            amls.reconcile()
+
+        vendor_bill = self.init_invoice(move_type='in_invoice', amounts=[1000], post=True)
+        payment = create_move_payment(vendor_bill, 10)
```

Comment:
The payment should be negative for vendor bills (or positive IDK, but it can't be the same sign as for customer invoices)

Reply:
That's because you need to take `amount = -partial.credit_amount_currency` to match the sign of `amount_signed`